### PR TITLE
refactor WalletModal.tsx component

### LINKED ISSUE #406: Connect wallet - mobile behavior
as a user, when i browse to the app using mobile phone and try to connect my wallet, the behavior should be updated to better support the wallet connections available.

1. if the user is directly coming to the app using a browser (chrome, safari, etc) -> display these wallet options:
- metamask -> when clicked -> open a popup with a message indicating they need to browse using the metamask browser. (details on the popup below)
- Wallet connect
- coinbase wallet
- fortmatic
2. if the user is using wallet browser (metamask for example) -> when clicking the wallet icon, login with the wallet they are using. one 2nd visit, remember the login and auto connect when they reach the site.

metamask mobile message:
title:
How to connect Metamask?
text: 
1. Open your metamask mobile application
2. Use the Browser option and navigate to bancor.network
3. Connect your wallet
Done!

mockup: TBD

### PR DESCRIPTION
**Pull request by pingustar**
fixes #406 #402 #329

ToDo:

- [x] refactor components
- [x] fix scrolling ... currently I removed the ability to open the wallet connect modal by reading the redux state, this fixes the scrolling but it is not a permanent solution. I dont know how to fix it - probably has to do with focus trap or something. Help wanted @zavelevsky @RanCohenn 

- [x] on mobile show only: MetaMask, Wallet Connect, Coinbase, Fortmatic
- [x] Autologin if MetaMask in-app browser is used
- [ ] Autologin if Coinbase / Fortmatic in-app browser is used
- [x] Show MetaMask message when not wallet in-app browser is used

@ashachaf what do you think about doing it like uniswap, they hide MetaMask on mobile and offer Wallet Connect. Wallet Connect brings you to the MetaMask app, you allow the connection, then go back to the browser, done.

@RanCohenn 
I am not sure what this does, do we need these two things from `tryConnecting` in `WalletModal.tsx`?

```ts
 if (
      connector instanceof WalletConnectConnector &&
      connector.walletConnectProvider?.wc?.uri
    )
      connector.walletConnectProvider = undefined;
```

and 

```ts
  if (error instanceof UnsupportedChainIdError) {
            activate(connector);
          } else setError(true);
        }
```

### CODE REVIEW

**Comment by RanCohenn** [comment]
@pingustar When first loading the app without being connected and logging in to metamask it doesnt show the metamask label in the pending view (just the logo) afterwards it is shown correctly

**Comment by RanCohenn** [comment]
No need to check the instance of the connector (old code)
Also no point in retrying after unsupported chain error (not sure why it was there)

File: src/elements/walletConnect/useWalletConnect.ts [hunk 1]
```diff
@@ -26,9 +29,15 @@ export const useWalletConnect = (): UseWalletConnect => {
   const { activate, deactivate, account, connector } = useWeb3React();
   const [isPending, setIsPending] = useState<boolean>(false);
   const [isError, setIsError] = useState<boolean>(false);
-  const [isOpen, setIsOpen] = useState<boolean>(false);
   const [selectedWallet, setSelectedWallet] = useState<WalletInfo>();
 
+  const isOpen = useAppSelector<boolean>((state) => state.user.walletModal);
+  const dispatch = useDispatch();
+
+  const setIsOpen = (value: boolean) => {
```
Comment (zavelevsky) [inline src/elements/walletConnect/useWalletConnect.ts:37]: please wrap these functions (setIsOpen, handleOpenModal, etc.) with useCallback

Comment (pingustar) [reply src/elements/walletConnect/useWalletConnect.ts]: done

File: src/elements/walletConnect/WalletConnectButton.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export const WalletConnectButton = ({
+  handleOpenModal,
+  handleDisconnect,
+  account,
+  selectedWallet,
+}: UseWalletConnect) => {
+  const buttonText = account ? shortenString(account) : 'Connect Wallet';
+
+  const handleClick = () => {
```
Comment (zavelevsky) [inline src/elements/walletConnect/WalletConnectButton.tsx:14]: useCallback

File: src/elements/walletConnect/WalletConnectModal.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Modal } from 'components/modal/Modal';
+import { UseWalletConnect } from './useWalletConnect';
+import { WalletConnectModalList } from './WalletConnectModalList';
+import { WalletConnectModalPending } from './WalletConnectModalPending';
+import { WalletConnectModalError } from './WalletConnectModalError';
+
+export const WalletConnectModal = (props: UseWalletConnect) => {
+  const { isPending, isError } = props;
+
+  const title = isError
```
Comment (zavelevsky) [inline src/elements/walletConnect/WalletConnectModal.tsx:10]: Best to either useMemo or take this function outside

File: src/elements/walletConnect/WalletConnectButton.tsx (hunk 2 above)
Comment (pingustar) [reply src/elements/walletConnect/WalletConnectButton.tsx]: done

File: src/elements/walletConnect/WalletConnectModal.tsx (hunk 3 above)
Comment (pingustar) [reply src/elements/walletConnect/WalletConnectModal.tsx]: done

File: src/components/modal/Modal.tsx [hunk 4]
```diff
@@ -28,18 +28,15 @@ export const Modal = ({
   onClose,
   large,
 }: ModalProps) => {
-  const refDiv = useRef(null);
-
   return (
     <>
       <Transition appear show={isOpen} as={Fragment}>
         <Dialog
-          initialFocus={refDiv}
```
Comment (zavelevsky) [inline src/components/modal/Modal.tsx:37]: focusing on the button adds a circle around it, need to fix

**Comment by ashachaf** [comment]
- [ ] x at the top right is highlighted when the modal opens
image

- [ ] wallet connect gets stuck sometimes. (try to open/close it several times in a row)
image

- [ ] mobile -> wallet connect + trust wallet. connection was done, i have received indication on trust wallet that i can go back to my browser and use the wallet. however, the browser didnt update and didnt establish the connection with wallet connect.

File: src/components/modal/Modal.tsx (hunk 4 above)
Comment (pingustar) [reply src/components/modal/Modal.tsx:37]: done

**Review by ashachaf — APPROVED** [review]

**Pull request merged**